qa/lucene-index-compatibility/src/javaRestTest/java/org/elasticsearch/lucene/SearchableSnapshotCompatibilityIT.java

```diff
@@ -0,0 +1,115 @@
+            .setting("xpack.searchable.snapshot.shared_cache.size", "16MB")
+            .setting("xpack.searchable.snapshot.shared_cache.region_size", "256KB");
+    }
+
+    public SearchableSnapshotCompatibilityIT(Version version) {
+        super(version);
+    }
+
+    public void testSearchableSnapshot() throws Exception {
+        final String repository = suffix("repository");
+        final String snapshot = suffix("snapshot");
+        final String index = suffix("index");
+        final int numDocs = 1234;
+
+        logger.debug("--> registering repository [{}]", repository);
+        registerRepository(
+            client(),
+            repository,
+            FsRepository.TYPE,
+            true,
+            Settings.builder().put("location", REPOSITORY_PATH.getRoot().getPath()).build()
+        );
+
+        if (VERSION_MINUS_2.equals(clusterVersion())) {
+            logger.debug("--> creating index [{}]", index);
+            createIndex(
+                client(),
+                index,
+                Settings.builder()
+                    .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                    .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+                    .put(IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey(), true)
+                    .build()
+            );
+
+            logger.debug("--> indexing [{}] docs in [{}]", numDocs, index);
+            final var bulks = new StringBuilder();
+            IntStream.range(0, numDocs).forEach(n -> bulks.append(Strings.format("""
+                {"index":{"_id":"%s","_index":"%s"}}
+                {"test":"test"}
+                """, n, index)));
+
+            var bulkRequest = new Request("POST", "/_bulk");
+            bulkRequest.setJsonEntity(bulks.toString());
+            var bulkResponse = client().performRequest(bulkRequest);
+            assertOK(bulkResponse);
+            assertThat(entityAsMap(bulkResponse).get("errors"), allOf(notNullValue(), is(false)));
+
+            logger.debug("--> creating snapshot [{}]", snapshot);
+            createSnapshot(client(), repository, snapshot, true);
+            return;
+        }
+
+        if (VERSION_MINUS_1.equals(clusterVersion())) {
+            ensureGreen(index);
+
+            assertThat(indexLuceneVersion(index), equalTo(VERSION_MINUS_2));
+            assertDocCount(client(), index, numDocs);
+
+            logger.debug("--> deleting index [{}]", index);
+            deleteIndex(index);
+            return;
+        }
+
+        if (VERSION_CURRENT.equals(clusterVersion())) {
+            var mountedIndex = suffix("index-mounted");
+            logger.fatal("--> mounting index [{}] as [{}]", index, mountedIndex);
+
+            // Mounting the index will fail as Elasticsearch does not support reading N-2 yet
+            var request = new Request("POST", "/_snapshot/" + repository + "/" + snapshot + "/_mount");
```

Comment:
Just a comment for later that we should also add a test where we mount the N-2 index on N-1 and then search it on N.

Reply:
I added a TODO :+1: